src/main/java/com/o19s/es/ltr/query/RankerQuery.java

```diff
@@ -252,6 +288,22 @@ public void normalize(float norm, float boost) {
         public RankerScorer scorer(LeafReaderContext context) throws IOException {
             List<Scorer> scorers = new ArrayList<>(weights.size());
             List<DocIdSetIterator> subIterators = new ArrayList<>(weights.size());
+
+            String pattern = "\\$(\\d+)";
```

Comment:
not a big fan of executing a regex here, why do we need to support this `$0` syntax?